mollie/orders.go

```diff
@@ -408,6 +409,7 @@ func (ors *OrdersService) CreateOrderPayment(orderID string, ordPay *OrderPaymen
 		if err != nil {
```

Comment:
This is redundant, we know that error is not nil, plus the error you get from client.Do() is already wrapped in the custom error response, I checked out your branch and that is the reason why tests are failing. 

The current code already handles gracefully the 422 case, so there is no need for an overhead.

If I am overlooking something please let me know, plus I cannot merge your branch until all the test cases are passing.

Reply:
Yes, it's the error check for the unmarshalling, it fails on HTTPRequestErrors test because there is nothing on res.content, just the header with Internal Server Error Status. That is why I check len(res.content) on my initial code of this PR to differentiate between the errors that have content or not. 
I can go and edit it like this:
```go
if res.ContentLength > 0 {
	err = json.Unmarshal(res.content, &errorResponse)
	if err != nil {
		return
	}
}
```
Per your suggestion https://github.com/VictorAvelar/mollie-api-go/pull/41#discussion_r339463738

Is that ok? Any suggestion?

Reply:
Yes, I think that will roll on, and then we can have the investigation issue to have a deeper look, as you're the one with more information about this issue I will be thankful if you can create the issue 😄

Reply:
Well, for me the issue here is just how to differentiate the response when it contains a content or not and the code already handled it now. I only encountered the error response structure for Unprocessable Entity. What if there is another error response structure in the same endpoint in the future?
So, for now I think we can leave it as it is :smile: